File: runtime/flutter/services/systemMouseCursors.ts
import { SystemMouseCursor } from "./systemMouseCursor";
declare const flutter: {
    services: {
        systemMouseCursorsNone: (this: void) => ISystemMouseCursor;
        systemMouseCursorsBasic: (this: void) => ISystemMouseCursor;
        systemMouseCursorsClick: (this: void) => ISystemMouseCursor;
        systemMouseCursorsForbidden: (this: void) => ISystemMouseCursor;
        systemMouseCursorsWait: (this: void) => ISystemMouseCursor;
        systemMouseCursorsProgress: (this: void) => ISystemMouseCursor;
        systemMouseCursorsContextMenu: (this: void) => ISystemMouseCursor;
        systemMouseCursorsHelp: (this: void) => ISystemMouseCursor;
        systemMouseCursorsText: (this: void) => ISystemMouseCursor;
        systemMouseCursorsVerticalText: (this: void) => ISystemMouseCursor;
        systemMouseCursorsCell: (this: void) => ISystemMouseCursor;
        systemMouseCursorsPrecise: (this: void) => ISystemMouseCursor;
        systemMouseCursorsMove: (this: void) => ISystemMouseCursor;
        systemMouseCursorsGrab: (this: void) => ISystemMouseCursor;
        systemMouseCursorsGrabbing: (this: void) => ISystemMouseCursor;
        systemMouseCursorsNoDrop: (this: void) => ISystemMouseCursor;
        systemMouseCursorsAlias: (this: void) => ISystemMouseCursor;
        systemMouseCursorsCopy: (this: void) => ISystemMouseCursor;
        systemMouseCursorsDisappearing: (this: void) => ISystemMouseCursor;
        systemMouseCursorsAllScroll: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeLeftRight: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeUpDown: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeUpLeftDownRight: (
            this: void
        ) => ISystemMouseCursor;
        systemMouseCursorsResizeUpRightDownLeft: (
            this: void
        ) => ISystemMouseCursor;
        systemMouseCursorsResizeUp: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeDown: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeLeft: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeRight: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeUpLeft: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeUpRight: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeDownLeft: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeDownRight: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeColumn: (this: void) => ISystemMouseCursor;
        systemMouseCursorsResizeRow: (this: void) => ISystemMouseCursor;
        systemMouseCursorsZoomIn: (this: void) => ISystemMouseCursor;
        systemMouseCursorsZoomOut: (this: void) => ISystemMouseCursor;
    };
};
export interface ISystemMouseCursors {}
export class SystemMouseCursors {
    public static none = flutter.services.systemMouseCursorsNone();
    public static basic = flutter.services.systemMouseCursorsBasic();
    public static click = flutter.services.systemMouseCursorsClick();
    public static forbidden = flutter.services.systemMouseCursorsForbidden();
    public static wait = flutter.services.systemMouseCursorsWait();
    public static progress = flutter.services.systemMouseCursorsProgress();
    public static contextMenu =
        flutter.services.systemMouseCursorsContextMenu();
    public static help = flutter.services.systemMouseCursorsHelp();
    public static text = flutter.services.systemMouseCursorsText();
    public static verticalText =
        flutter.services.systemMouseCursorsVerticalText();
    public static cell = flutter.services.systemMouseCursorsCell();
    public static precise = flutter.services.systemMouseCursorsPrecise();
    public static move = flutter.services.systemMouseCursorsMove();
    public static grab = flutter.services.systemMouseCursorsGrab();
    public static grabbing = flutter.services.systemMouseCursorsGrabbing();
    public static noDrop = flutter.services.systemMouseCursorsNoDrop();
    public static alias = flutter.services.systemMouseCursorsAlias();
    public static copy = flutter.services.systemMouseCursorsCopy();
    public static disappearing =
        flutter.services.systemMouseCursorsDisappearing();
    public static allScroll = flutter.services.systemMouseCursorsAllScroll();
    public static resizeLeftRight =
        flutter.services.systemMouseCursorsResizeLeftRight();
    public static resizeUpDown =
        flutter.services.systemMouseCursorsResizeUpDown();
    public static resizeUpLeftDownRight =
        flutter.services.systemMouseCursorsResizeUpLeftDownRight();
    public static resizeUpRightDownLeft =
        flutter.services.systemMouseCursorsResizeUpRightDownLeft();
    public static resizeUp = flutter.services.systemMouseCursorsResizeUp();
    public static resizeDown = flutter.services.systemMouseCursorsResizeDown();
    public static resizeLeft = flutter.services.systemMouseCursorsResizeLeft();
    public static resizeRight =
        flutter.services.systemMouseCursorsResizeRight();
    public static resizeUpLeft =
        flutter.services.systemMouseCursorsResizeUpLeft();
    public static resizeUpRight =
        flutter.services.systemMouseCursorsResizeUpRight();
    public static resizeDownLeft =
        flutter.services.systemMouseCursorsResizeDownLeft();
    public static resizeDownRight =
        flutter.services.systemMouseCursorsResizeDownRight();
    public static resizeColumn =
        flutter.services.systemMouseCursorsResizeColumn();
    public static resizeRow = flutter.services.systemMouseCursorsResizeRow();
    public static zoomIn = flutter.services.systemMouseCursorsZoomIn();
    public static zoomOut = flutter.services.systemMouseCursorsZoomOut();
}

File: lib/cfr/builtins/libs/flutter/services/systemMouseCursors.dart
import 'dart:core' as _fac9;

import 'package:flutter/src/services/mouse_cursor.dart' as _fd33;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedSystemMouseCursors extends _36c2.VMManagedBox<_fd33.SystemMouseCursors> {VMManagedSystemMouseCursors({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {  }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _fd33.SystemMouseCursors vmObject;

 }

void loadSystemMouseCursors({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'systemMouseCursorsNone'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.none, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsBasic'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.basic, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsClick'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.click, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsForbidden'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.forbidden, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsWait'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.wait, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsProgress'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.progress, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsContextMenu'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.contextMenu, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsHelp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.help, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsText'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.text, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsVerticalText'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.verticalText, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsCell'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.cell, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsPrecise'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.precise, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsMove'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.move, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsGrab'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.grab, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsGrabbing'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.grabbing, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsNoDrop'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.noDrop, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsAlias'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.alias, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsCopy'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.copy, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsDisappearing'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.disappearing, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsAllScroll'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.allScroll, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeLeftRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeLeftRight, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeUpDown'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeUpDown, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeUpLeftDownRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeUpLeftDownRight, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeUpRightDownLeft'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeUpRightDownLeft, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeUp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeUp, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeDown'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeDown, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeLeft'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeLeft, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeRight, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeUpLeft'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeUpLeft, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeUpRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeUpRight, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeDownLeft'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeDownLeft, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeDownRight'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeDownRight, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeColumn'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeColumn, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsResizeRow'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.resizeRow, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsZoomIn'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.zoomIn, hydroState: hydroState, table: _36c2.HydroTable())]; } );
table  [
'systemMouseCursorsZoomOut'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return  [_36c2.maybeBoxObject<_fd33.SystemMouseCursor>(object: _fd33.SystemMouseCursors.zoomOut, hydroState: hydroState, table: _36c2.HydroTable())]; } );
_36c2.registerBoxer<_fd33.SystemMouseCursors>(boxer: ({required _fd33.SystemMouseCursors vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedSystemMouseCursors(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
